src/ray/core_worker/test/core_worker_test.cc

```diff
@@ -357,9 +362,9 @@ void CoreWorkerTest::TestActorReconstruction(
         RAY_LOG(INFO) << "killing worker";
         ASSERT_EQ(system("pkill mock_worker"), 0);
 
-        // Wait for actor restruction event, and then for alive event.
-        ASSERT_TRUE(
-            WaitForDirectCallActorState(driver, actor_id, false, 30 * 1000 /* 30s */));
+        // Sleep for 1s while the actor is restarted, then wait for it to
+        // become alive again.
+        usleep(1000 * 1000);
```

Comment:
The previous test was wrong and I believe it only worked because it was slow. It waits for a RECONSTRUCTING state, then an ALIVE state, but the actor can switch through these states correctly before the next time the condition is checked. Then, the first check for RECONSTRUCTING will hang.

Reply:
Sorry that it's hacky, but I would have preferred to not have this test at all (we should be testing these things from python, not here). Just trying to fix it up enough that it passes right now.